airbyte-webapp/src/hooks/services/ConnectionEdit/ConnectionEditService.tsx

```diff
@@ -11,22 +17,69 @@ interface ConnectionEditProps {
   connectionId: string;
 }
 
-const useConnectionEdit = ({ connectionId }: ConnectionEditProps) => {
-  const [connection, setConnection] = useState(useGetConnection(connectionId));
+export interface ConnectionCatalog {
+  syncCatalog: AirbyteCatalog;
+  catalogId?: string;
+}
+
+interface ConnectionEditHook {
+  connection: WebBackendConnectionRead;
+  connectionUpdating: boolean;
+  schemaError?: Error;
+  schemaRefreshing: boolean;
+  schemaHasBeenRefreshed: boolean;
+  updateConnection: (connectionUpdates: WebBackendConnectionUpdate) => Promise<void>;
+  refreshSchema: () => Promise<void>;
+  discardRefreshedSchema: () => void;
+}
+
+const getConnectionCatalog = (connection: WebBackendConnectionRead): ConnectionCatalog =>
+  pick(connection, ["syncCatalog", "catalogId"]);
+
+const useConnectionEdit = ({ connectionId }: ConnectionEditProps): ConnectionEditHook => {
   const connectionService = useWebConnectionService();
+  const [connection, setConnection] = useState(useGetConnection(connectionId));
+  const [catalog, setCatalog] = useState<ConnectionCatalog>(() => getConnectionCatalog(connection));
   const [schemaHasBeenRefreshed, setSchemaHasBeenRefreshed] = useState(false);
 
   const [{ loading: schemaRefreshing, error: schemaError }, refreshSchema] = useAsyncFn(async () => {
     const refreshedConnection = await connectionService.getConnection(connectionId, true);
-    setConnection(refreshedConnection);
-    setSchemaHasBeenRefreshed(true);
+    if (refreshedConnection.catalogDiff && refreshedConnection.catalogDiff.transforms?.length > 0) {
```

Comment:
If we're only sometimes updating this then I think we should have a toast or something.
"Source had no changes" and "Source schema refreshed" perhaps

Reply:
If there are diffs, then the diff modal will appear.

If there are no changes, It may be good to toast letting the user know.